assets/src/application/services/toaster/useSystemNotificationsManager.tsx

```diff
@@ -0,0 +1,66 @@
+import { useCallback } from 'react';
+
+// import useLoadingToast from './useLoadingToast';
+import useToaster from './useToaster';
+import { AddToast, DissmissToast, SystemNotificationsManager, TOAST_STATUS } from './types';
+
+const useSystemNotificationsManager: SystemNotificationsManager = (options?) => {
+	const { addToast, dissmiss: dissmissToast, dissmissAll, exists, generateKey, getToasts } = useToaster(options);
+
+	const dismiss: DissmissToast = useCallback((key) => dissmissToast(key), [dissmissToast]);
+
+	const error: AddToast = useCallback(
+		({ key, ...props }): string => {
+			return key && exists(key) ? key : addToast({ ...props, type: TOAST_STATUS.ERROR });
+		},
+		[addToast]
+	);
+
+	const info: AddToast = useCallback(
+		({ key, ...props }): string => {
+			return key && exists(key) ? key : addToast({ ...props, type: TOAST_STATUS.INFO });
+		},
+		[addToast]
+	);
+
+	const loading = useCallback(
+		({ key, loading, ...props }): string => {
+			if (loading) {
+				return key && exists(key) ? key : addToast({ ...props, type: TOAST_STATUS.LOADING });
+			}
+			if (key) {
+				dissmissToast(key);
+				return 'dissmissed';
+			}
+			return 'loading...';
+		},
+		[addToast, dissmissToast]
+	);
+
+	const success: AddToast = useCallback(
+		({ duration = 30000, key, ...props }): string => {
+			return key && exists(key) ? key : addToast({ ...props, duration, type: TOAST_STATUS.SUCCESS });
+		},
+		[addToast]
+	);
+
+	const warning: AddToast = useCallback(
+		({ key, ...props }): string => {
+			return key && exists(key) ? key : addToast({ ...props, type: TOAST_STATUS.WARNING });
+		},
+		[addToast]
+	);
```

Comment:
there is a bit of duplication here, might want to refactor this a bit later via a factory function or similar, not strictly related to the current PR's scope

Reply:
ya that's a good idea